mantle/platform/qemu.go

```diff
@@ -743,12 +743,12 @@ func (builder *QemuBuilder) AddDisk(disk *Disk) error {
 
 // AddInstallIso adds an ISO image
 func (builder *QemuBuilder) AddInstallIso(path string) error {
-	// We use bootindex=2 here: the idea is that during an ISO install, the
+	// We use the cdrom option here: the idea is that during an ISO install, the
 	// primary disk isn't bootable, so the bootloader will fall back to the ISO
 	// boot. On reboot when the system is installed, the primary disk is
 	// selected. This allows us to have "boot once" functionality on both UEFI
 	// and BIOS (`-boot once=d` OTOH doesn't work with OVMF).
-	builder.Append("-drive", "file="+path+",format=raw,if=none,readonly=on,id=installiso", "-device", "ide-cd,drive=installiso,bootindex=2")
+	builder.Append("-cdrom", path)
```

Comment:
But we need the `bootindex` bits here right?  I think we need instead to have arch-dependent devices for cdrom or so.

Reply:
i thought the primary purpose of setting bootindex to 2 was to make sure that this device does not get picked first on a reboot. With the `-cdrom` option, it always first defaults to the primary disk anyway, so i thought that not necessary.

Reply:
See https://github.com/coreos/coreos-assembler/pull/1312
and it's notable that this PR is failing in the 4k sector test which is using UEFI.  It might be that we're doing CDROM first on UEFI or something?

Reply:
ah..thanks for that context...yes uefi breaks with this change..it doesn't even recognize that cdrom as bootable.i'll dig in more or we might need to have two paths - one for uefi and one for bios

Reply:
just made cdrom specific to s390x/ppc64le as those don't support uefi anyway.